addons/package_hierarchy/models/package_links.py

```diff
@@ -52,6 +52,57 @@ class PackageHierarchyLink(models.Model):
     )
     company_id = fields.Many2one("res.company", default=lambda self: self.env.company)
 
+    @api.model
+    def create(self, vals):
+        """
+        Extend create to prevent creating duplicate parent/child relations
+        and instead return the duplicate. This is in place of a constraint
+        as it is more user friendly
+        (instead of raising an error will simply return the existing link)
+        """
+        PackageHierarchyLink = self.env["package.hierarchy.link"]
+        move_line_ids = vals.get("move_line_ids")
+        if move_line_ids:
+            move_line_ids = move_line_ids[0][2]
+        existing_link = PackageHierarchyLink.search_read(
+            [
+                ("parent_id", "=", vals.get("parent_id")),
+                ("child_id", "=", vals.get("child_id")),
+                ("move_line_ids", "=", move_line_ids if move_line_ids else False),
```

Comment:
don't think we need a `if move_line_ids else False`, as `vals.get()` returns `None`, which I think has the same affect as False during the `search_read` :slightly_smiling_face:

Reply:
Because move_line_ids is a m2m it's `[[6, 0, []]` if no ids and passing `[]` to the domain doesn't work

Reply:
Oh I see